mono/eglib/glib.h

```diff
@@ -107,6 +107,56 @@ struct g_cast
 
 #endif
 
+#ifdef __cplusplus
+
+// Provide for bit operations on enums, but not all integer operations.
+// This alleviates a fair number of casts in porting C to C++.
+
+// Forward declare template with no generic implementation.
+template <size_t> struct mono_size_to_int;
+
+// Template specializations.
+template <> struct mono_size_to_int<1> { typedef int8_t T; };
```

Comment:
C++ type traits libraries usually define the result type with name `type`, not `T`.

Reply:
I was shielding developers from the verbosity, and think it is reasonable, but will fix.